src/main/kotlin/com/doist/gradle/changelog/GradleApiExt.kt

```diff
@@ -0,0 +1,13 @@
+package com.doist.gradle.changelog
+
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import java.io.File
+
+internal fun DirectoryProperty.asPendingChangelogDir(): File? {
+    val dir = asFile.orNull ?: return null
+    if (!dir.exists()) return null
+    if (!dir.isDirectory) throw GradleException("${dir.path} is not a directory.")
+
+    return dir
```

Comment:
Nit. I think `when` would be more idiomatic here :)

Reply:
Or we could replace 
```
val dir = asFile.orNull ?: return null
if (!dir.exists()) return null
```
with
```
val dir = asFile.orNull?.takeIf { it.exists() } ?: return null
```
😊